website/server/controllers/api-v3/webhook.js

```diff
@@ -0,0 +1,193 @@
+import { authWithHeaders } from '../../middlewares/auth';
+import { model as Webhook } from '../../models/webhook';
+import { removeFromArray } from '../../libs/collectionManipulators';
+import { NotFound } from '../../libs/errors';
+
+let api = {};
+
+/**
+* @api {post} /api/v3/user/webhook Create a new webhook - BETA
+* @apiName UserAddWebhook
+* @apiGroup Webhook
+*
+* @apiParam {UUID} [id="Randomly Generated UUID"] Body parameter - The webhook's id
+* @apiParam {String} url Body parameter - The webhook's URL
+* @apiParam {Boolean} [enabled=true] Body parameter - If the webhook should be enabled
+* @apiParam {Sring="taskActivity","groupChatReceived"} [type="taskActivity"] Body parameter - The webhook's type.
+* @apiParam {Object} [options] Body parameter - The webhook's options. Wil differ depending on type. Required for `groupChatReceived` type. If a webhook supports options, the dfault values are displayed in the examples below
+* @apiParamExample {json} Basic Example (Task Scored)
+*   {
+*     "enabled": true,
+*     "url": "http://some-webhook-url.com"
+*   }
+* @apiParamExample {json} Task Scored With Specified Id Example
+*   {
+*     "id": "a-valid-uuid-goes-here",
+*     "enabled": true,
+*     "url": "http://some-webhook-url.com",
+*     "type": "taskActivity"
+*   }
+* @apiParamExample {json} Task Activity Example
+*   {
+*     "enabled": true,
+*     "url": "http://some-webhook-url.com",
+*     "type": "taskActivity",
+*     "options": {
+*       "created": false,
+*       "updated": false,
+*       "deleted": false,
+*       "scored": true
+*     }
+*   }
+* @apiParamExample {json} Group Chat Received Example
+*   {
+*     "enabled": true,
+*     "url": "http://some-webhook-url.com",
+*     "type": "groupChatReceived",
+*     "options": {
+*       "groupId": "required-uuid-of-group"
+*     }
+*   }
+*
+* @apiSuccess {Object} data The created webhook
+* @apiSuccess {UUID} data.id The uuid of the webhook
+* @apiSuccess {String} data.url The url of the webhook
+* @apiSuccess {Boolean} data.enabled Whether the webhook should be sent
+* @apiSuccess {String} data.type The type of the webhook
+* @apiSuccess {Object} data.options The options for the webhook (See examples)
+*
+* @apiError InvalidUUID The `id` was not a valid `UUID`
+* @apiError InvalidEnable The `enable` param was not a `Boolean` value
+* @apiError InvalidUrl The `url` param was not valid url
+* @apiError InvalidWebhookType The `type` param was not a supported Webhook type
+* @apiError GroupIdIsNotUUID The `options.groupId` param is not a valid UUID for groupChatReceived webhook type
+*/
+api.addWebhook = {
+  method: 'POST',
+  middlewares: [authWithHeaders()],
+  url: '/user/webhook',
+  async handler (req, res) {
+    let user = res.locals.user;
+    let webhook = new Webhook(req.body);
+
+    await webhook.validate();
```

Comment:
I _think_ that being webhooks under user they get validated when the user is validated so this would be a repetition. If it's needed I'd at least use validateSync that is much faster